src/authentication/BearerWebIdExtractor.ts

```diff
@@ -25,13 +24,18 @@ export class BearerWebIdExtractor extends CredentialsExtractor {
     }
   }
 
-  public async handle(request: HttpRequest): Promise<CredentialSet> {
+  public async handle(request: HttpRequest): Promise<Credentials> {
     const { headers: { authorization }} = request;
 
     try {
-      const { webid: webId } = await this.verify(authorization!);
-      this.logger.info(`Verified WebID via Bearer access token: ${webId}`);
-      return { [CredentialGroup.agent]: { webId }};
+      const { webid: webId, client_id: clientId, iss: issuer } = await this.verify(authorization!);
+      this.logger.info(`Verified credentials via Bearer access token. WebID: ${webId
+      }, client ID: ${clientId}, issuer: ${issuer}`);
+      const credentials: Credentials = { agent: { webId }, issuer: { url: issuer }};
+      if (clientId) {
+        credentials.client = { webId: clientId };
+      }
+      return credentials;
```

Comment:
```suggestion
      return {
        agent: { webId },
        issuer: { url: issuer },
        client: clientId && { webId: clientId },
      };
```
or, with variable renaming,
```suggestion
      return {
        agent: { webId: agent },
        issuer: { url: issuer },
        client: client && { webId: client },
      };
```

Reply:
Doesn't work since if `clientId` is an empty string it would try to assign that string to the `client` field since an empty string is a falsy value.

Doing something like `client: typeof clientId === 'undefined' ? clientId : { webId: clientId }` would work. But personally I prefer this solution as it doesn't put keys in the object unless they have a defined value.